src/qibolab/compilers/compiler.py

```diff
@@ -120,13 +119,12 @@ def _compile_gate(
 
         # update global pulse sequence
         # determine the right start time based on the availability of the qubits involved
-        all_qubits = {*gate_sequence.qubits, *gate.qubits}
+        all_qubits = {*gate_sequence.qubits, *gate.qubits, *gate_phases.keys()}
```

Comment:
Is there any particular reason `gate_phases` is needed here. I would expect that for our applications this is always a subset of `gate.qubits` so it shouldn't make any difference, but even if there is no, why do we need to take these qubits into account for determining the start time?

Reply:
If I understood correctly, `gate.qubits` are always numeric indices, but platform qubit names can be strings as well, so it could be that in the pulse sequence none of `gate.qubits` is present, because sequence uses the platform qubit names.

Given what I said above is true, then I added phase qubits here having in mind two-qubit gates that are done through a coupler. Imagine a simple gate that consists of only a coupler pulse (and virtual phases, but no pulse on the qubits). And let's say the qubits already have a lot pulses, but the coupler does not have any yet. We want to add the coupler pulse after all qubit pulses, and not in the beginning of the coupler channel.